### PR TITLE
Validate port range

### PR DESCRIPTION
**Pull request by kebe7jun**
Port can not be 0.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[X] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: pilot/pkg/kube/inject/inject.go [hunk 1]
```diff
@@ -311,6 +311,9 @@ func parsePort(portStr string) (int, error) {
 	if err != nil {
 		return 0, fmt.Errorf("failed parsing port '%d': %v", port, err)
 	}
+	if port == 0 {
+		return 0, fmt.Errorf("failed parsing port '%d': port can not be 0", port)
```
Comment (nmittler) [inline pilot/pkg/kube/inject/inject.go]: nit: you're saying it's `0` twice in the log

**Comment by kebe7jun** [comment]
@howardjohn Ok, I see, I'll close this re.

**Pull request closed without merging**